java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java

```diff
@@ -41,129 +37,91 @@
-
-  // Response URL Template with placeholders for space id.
-  private static final String RESPONSE_URL_TEMPLATE =
-      "https://chat.googleapis.com/v1/__SPACE_ID__/messages";
 
-  // Response echo message template.
-  private static final String RESPONSE_TEMPLATE = "You said: `__MESSAGE__`";
+  /**
+   * A demo bot which implements {@link MessageReceiver} to receive messages. It simply echoes the
+   * incoming messages.
+   */
+  static class EchoBot implements MessageReceiver {
 
-  private static final String ADDED_RESPONSE = "Thank you for adding me!";
-
-  GoogleCredential credential;
-  HttpTransport httpTransport;
-  HttpRequestFactory requestFactory;
-
-  EchoBot() throws Exception {
-    credential =
-        GoogleCredential.fromStream(new FileInputStream(SERVICE_ACCOUNT_KEY_PATH))
-            .createScoped(Collections.singleton(HANGOUTS_CHAT_API_SCOPE));
-    httpTransport = GoogleNetHttpTransport.newTrustedTransport();
-    requestFactory = httpTransport.createRequestFactory(credential);
-  }
-
-  // Called when a message is received by the subscriber.
-  @Override
-  public void receiveMessage(PubsubMessage pubsubMessage, AckReplyConsumer consumer) {
-    System.out.println("Id : " + pubsubMessage.getMessageId());
-    // handle incoming message, then ack/nack the received message
-    try {
-      ObjectMapper mapper = new ObjectMapper();
-      JsonNode dataJson = mapper.readTree(pubsubMessage.getData().toStringUtf8());
-      System.out.println("Data : " + dataJson.toString());
-      handle(dataJson);
-      consumer.ack();
-    } catch (Exception e) {
-      System.out.println(e);
-      consumer.nack();
+    EchoBot() {
     }
-  }
-
-  public void handle(JsonNode eventJson) throws Exception {
-    JsonNodeFactory jsonNodeFactory = new JsonNodeFactory(false);
-    ObjectNode responseNode = jsonNodeFactory.objectNode();
 
-    // Construct the response depending on the event received.
-
-    String eventType = eventJson.get("type").asText();
-    switch (eventType) {
-      case "ADDED_TO_SPACE":
-        responseNode.put("text", ADDED_RESPONSE);
-        // A bot can also be added to a room by @mentioning it in a message. In that case, we fall
-        // through to the MESSAGE case and let the bot respond. If the bot was added using the
-        // invite flow, we just post a thank you message in the space.
-        if(!eventJson.has("message")) {
-          break;
+    // Called when a message is received by the subscriber.
+    @Override
+    public void receiveMessage(PubsubMessage pubsubMessage, AckReplyConsumer consumer) {
+      System.out.println("Id : " + pubsubMessage.getMessageId());
+      // handle incoming message, then ack/nack the received message
+      try {
+        String content = pubsubMessage.getData().toStringUtf8();
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode chatEvent = mapper.readTree(content);
+
+        Message reply = null;
```

Comment:
let's refactor to later check a boolean or initialize to an empty message and check if content present later